zk.js/src/test-utils/test-rpc.ts

```diff
@@ -195,14 +212,18 @@ export class TestRpc extends Rpc {
       if (this.indexedTransactions.length === 0) return [];
 
       const mostRecentTransaction = this.indexedTransactions.reduce((a, b) =>
-        a.transaction.blockTime > b.transaction.blockTime ? a : b,
+        (a.transaction as ParsedIndexedTransaction).blockTime >
```

Comment:
let's try to not cast the type since this overrides all typescript checks and can lead to runtime errors; instead we can handle this using a type guard:

```
function isParsedIndexedTransaction(
      transaction: any,
    ): transaction is ParsedIndexedTransaction {
      return "blockTime" in transaction;
    }
```
    
    
e.g.:    
  ```
 const mostRecentTransaction = this.indexedTransactions.reduce((a, b) => {
      const aBlockTime = isParsedIndexedTransaction(a.transaction)
        ? a.transaction.blockTime
        : 0;
      const bBlockTime = isParsedIndexedTransaction(b.transaction)
        ? b.transaction.blockTime
        : 0;
      return aBlockTime > bBlockTime ? a : b;
    });
```

Reply:
the same for all other uses of type casting (used heavily in the other fns) - pls replace with type guards.  maybe it makes sense to look at the underlying root cause of why we have incoherent types for thes fns and resolve that if possible